components/reference-data/src/test/java/org/datacleaner/beans/transform/RemoveDictionaryMatchesTransformerTest.java

```diff
@@ -74,31 +74,30 @@ public void testJobTitleScenarioRemovedMatchesAsString() throws Throwable {
         assertEquals("Software Engineer", transformer.transform("Senior Software Engineer")[0]);
         assertEquals("Senior", transformer.transform("Senior Software Engineer")[1]);
         
-        assertEquals("Designer", transformer.transform("  Lead   Designer  ")[0]);
+        assertEquals("    Designer  ", transformer.transform("  Lead   Designer  ")[0]);
         assertEquals("Lead", transformer.transform("  Lead   Designer  ")[1]);
         
-        assertEquals("Software Engineer", transformer.transform("Software  Engineer")[0]);
+        assertEquals("Software  Engineer", transformer.transform("Software  Engineer")[0]);
         assertEquals("", transformer.transform("Software  Engineer")[1]);
         
         assertEquals("Guru of employees", transformer.transform("Principal Senior Lead Guru of Junior employees")[0]);
-        assertEquals("Principal Senior Lead Junior", transformer.transform("Principal Senior Lead Guru of Junior employees")[1]);
+        assertEquals("Principal Junior Senior Lead", transformer.transform("Principal Senior Lead Guru of Junior employees")[1]);
    
     }
     public void testJobTitleScenarioRemovedMatchesAsList() throws Throwable {
-    
         transformer._removedMatchesType= RemovedMatchesType.LIST;
         
         assertEquals("Software Engineer", transformer.transform("Senior Software Engineer")[0]);
         assertEquals("[Senior]", transformer.transform("Senior Software Engineer")[1].toString());
         
-        assertEquals("Designer", transformer.transform("  Lead   Designer  ")[0]);
+        assertEquals("    Designer  ", transformer.transform("  Lead   Designer  ")[0]);
         assertEquals("[Lead]", transformer.transform("  Lead   Designer  ")[1].toString());
         
-        assertEquals("Software Engineer", transformer.transform("Software  Engineer")[0]);
+        assertEquals("Software  Engineer", transformer.transform("Software  Engineer")[0]);
         assertEquals("[]", transformer.transform("Software  Engineer")[1].toString());
         
         assertEquals("Guru of employees", transformer.transform("Principal Senior Lead Guru of Junior employees")[0]);
-        assertEquals("[Principal, Senior, Lead, Junior]", transformer.transform("Principal Senior Lead Guru of Junior employees")[1].toString());
+        assertEquals("[Principal, Junior, Senior, Lead]", transformer.transform("Principal Senior Lead Guru of Junior employees")[1].toString());
```

Comment:
I'm not really a fan of keeping the old way around; There's other changes in behaviour (i.e. what a boundary character is), and it messes with the string by tokenizing and stitching it together again.

I'll think a bit about how to retain it... Including if we even want to (somehow "I replaced them in this order" is more important than "I received them in this order" to me, but I see the compatibility part).